jvm/src/main/scala/async/VThreadSupport.scala

```diff
@@ -5,12 +5,17 @@ import java.util.concurrent.locks.ReentrantLock
 import scala.concurrent.duration.FiniteDuration
 
 object VThreadScheduler extends Scheduler:
-  override def execute(body: Runnable): Unit = Thread.startVirtualThread(body)
+  private val VTFactory = Thread.ofVirtual()
+    .name("gears.async.VThread-", 0L)
+    .factory()
+  
+  override def execute(body: Runnable): Unit = VTFactory.newThread(body)
 
   override def schedule(delay: FiniteDuration, body: Runnable): Cancellable =
-    val th = Thread.startVirtualThread: () =>
+    val th = VTFactory.newThread: () =>
       Thread.sleep(delay.toMillis)
-      execute(body)
+      body.run()
```

Comment:
We changed this a while ago as this would cause the `body` to be cancelled when the sleep is cancelled after the `Thread.sleep`  ;)

Reply:
Perhaps we could add some kind of indicator that sleeping has already completed, so the cancellation is void after that.